### DISCUSSION TITLE
Unable to change power control settings ST5000LM000-2AN170

### DISCUSSION THREAD

**Discussion by Baswazz** [Seagate/openSeaChest#146] [Q&A] [answered]
I am unable to get my dives to spindown. The drives are Seagate Barracuda SATA 2.5 5TB [ST5000LM000-2AN170](https://www.seagate.com/support/internal-hard-drives/laptop-hard-drives/barracuda-2-5/).
I get: Showing EPC Settings not supported on this device.

I checked if there where any firmware updates: No firmware updates currently available for serial number: WCJ8RV4B

`./openSeaChest_PowerControl --showEPCSettings -d all`

```
==========================================================================================
 openSeaChest_PowerControl - openSeaChest drive utilities - NVMe Enabled
 Copyright (c) 2014-2024 Seagate Technology LLC and/or its Affiliates, All Rights Reserved
 openSeaChest_PowerControl Version: 3.6.0-7_1_0 ARM64
 Build Date: May 21 2024
 Today: Tue May 21 20:51:59 2024	User: root
==========================================================================================

/dev/sg0 - ST5000LM000-2AN170 - WCJ8RV4B - 0001 - ATA
Showing EPC Settings not supported on this device.

/dev/sg1 - ST5000LM000-2AN170 - WCJ8J2EV - 0001 - ATA
Showing EPC Settings not supported on this device.

/dev/sg2 - ST5000LM000-2AN170 - WCJ8S94G - 0001 - ATA
Showing EPC Settings not supported on this device.

/dev/sg3 - ST5000LM000-2AN170 - WCJ8S919 - 0001 - ATA
Showing EPC Settings not supported on this device.
```

I was able to show the APM and set the APM level from 128 to 127.
`sudo ./openSeaChest_PowerControl --showAPMLevel -d all`
`sudo ./openSeaChest_PowerControl --setAPMLevel 127 -d all`

```
/dev/sg0 - ST5000LM000-2AN170 - WCJ8RV4B - 0001 - ATA
Current APM Level is 127 (Intermediate power management with Standby mode)

/dev/sg1 - ST5000LM000-2AN170 - WCJ8J2EV - 0001 - ATA
Current APM Level is 127 (Intermediate power management with Standby mode)

/dev/sg2 - ST5000LM000-2AN170 - WCJ8S94G - 0001 - ATA
Current APM Level is 127 (Intermediate power management with Standby mode)

/dev/sg3 - ST5000LM000-2AN170 - WCJ8S919 - 0001 - ATA
Current APM Level is 127 (Intermediate power management with Standby mode)
```

Only thing the drives seem todo is PM0 or PM1:
`sudo ./openSeaChest_PowerControl --checkPowerMode -d all`

```
/dev/sg0 - ST5000LM000-2AN170 - WCJ8RV4B - 0001 - ATA
Device is in the PM0: Active state or PM1: Idle State

/dev/sg1 - ST5000LM000-2AN170 - WCJ8J2EV - 0001 - ATA
Device is in the PM0: Active state or PM1: Idle State

/dev/sg2 - ST5000LM000-2AN170 - WCJ8S94G - 0001 - ATA
Device is in the PM0: Active state or PM1: Idle State

/dev/sg3 - ST5000LM000-2AN170 - WCJ8S919 - 0001 - ATA
Device is in the PM0: Active state or PM1: Idle State
```

Drive info:
`sudo ./openSeaChest_Info -d /dev/sg0 --deviceInfo`

```
/dev/sg0 - ST5000LM000-2AN170 - WCJ8RV4B - 0001 - ATA
	Model Number: ST5000LM000-2AN170
	Serial Number: WCJ8RV4B
	Firmware Revision: 0001
	World Wide Name: 5000C500F2091FF4
	Drive Capacity (TB/TiB): 5.00/4.55
	Native Drive Capacity (TB/TiB): 5.00/4.55
	Temperature Data:
		Current Temperature (C): 39
		Highest Temperature (C): 53
		Lowest Temperature (C): 26
	Power On Time:  7 hours 55 minutes
	Power On Hours: 7.92
	MaxLBA: 9767541167
	Native MaxLBA: 9767541167
	Logical Sector Size (B): 512
	Physical Sector Size (B): 4096
	Sector Alignment: 0
	Rotation Rate (RPM): 5526
	Form Factor: 2.5"
	Last DST information:
		DST has never been run
	Long Drive Self Test Time:  13 hours 34 minutes
	Interface speed:
		Max Speed (Gb/s): 6.0
		Negotiated Speed (Gb/s): 6.0
	Annualized Workload Rate (TB/yr): 41.83
	Total Bytes Read (MB): 263.75
	Total Bytes Written (GB): 37.54
	Encryption Support: Full Disk Encryption
	Cache Size (MiB): 128.00
	Read Look-Ahead: Enabled
	Write Cache: Enabled
	Low Current Spinup: Enabled
	SMART Status: Good
	ATA Security Information: Supported
	Firmware Download Support: Full, Segmented, Deferred
	Specifications Supported:
		ACS-3
		ACS-2
		ATA8-ACS
		ATA/ATAPI-7
		ATA/ATAPI-6
		ATA/ATAPI-5
		ATA/ATAPI-4
		ACS-3 (Revision 5)
		SATA 3.1
		SATA 3.0
		SATA 2.6
		SATA 2.5
		SATA II: Extensions
		SATA 1.0a
		ATA8-AST
	Features Supported:
		Read/Write Multiple
		SATA NCQ [QD=32]
		SATA Software Settings Preservation [Enabled]
		SATA Device Initiated Power Management
		HPA
		Power Management
		Security
		SMART [Enabled]
		DCO
		48bit Address
		Set Max Security Extension
		PUIS [Enabled]
		APM [Enabled]
		GPL
		SMART Self-Test
		SMART Error Logging
		Sense Data Reporting [Enabled]
		SCT Write Same
		SCT Data Tables
		Host Logging
		Seagate In Drive Diagnostics (IDD)
	Adapter Information:
		Adapter Type: PCI
		Vendor ID: 197Bh
		Product ID: 0585h
		Revision: 0000h
```

**Comment by Baswazz** [comment]
I have an update, setting the APM level to 127 was enough to spindown the drives. I do not know what the default spindown time is for these drives and it looks like the firmware does not allow me to set a custom spindown time.
This also works with `sudo hdparm -B /dev/sd[abcd]`

**Comment by ThunderEX** [comment]
I think it is expected.
APM is a legacy feature which only allows you to set single APM level, with all detailed timers controlled by firmware according to APM level setting.
EPC is a modern feature which allows customization on all power mode timers.
ST5000LM000 is an old product launched in 8 years ago. And as `-i` output shows, it only support APM but not EPC.

**Comment by vonericsen** [comment, marked as answer]
@Baswazz,

On drives with EPC (Extended power conditions), it's easy to know what the specific timing for idle/standby is, but on older drives it is vendor unique what the timers are set to and there are some interactions between the drive's standby timer and the last change to APM (Advanced power management) that can be a little confusing.

Old drives won't have EPC and it is unlikely to be made available with a firmware update.

Before talking about APM, old ATA drives do have a way to specify the value of the standby timer. The only exception is really, really old drives before ATA standardization or only ATA-1 compliance. For these old drives, all behavior is vendor unique and you will need to product manual to figure out how to do this.

In openSeaChest_PowerControl, you can use the option `--standby` followed by a time in milliseconds to set the amount of time the drive must be idle before going into standby (spun-down). This time in milliseconds is converted according to the SAT spec's formula to map it to the values the standby/idle commands expect. 
These old commands can set 5 second increments up to 20 minutes, 30 minute increments between 30 minutes and 5.5 hours, 21min, some time between 8h and 12h, or 21mins 15s.
This is why we use the SAT spec formula to convert to the most sensible value since there are some very specific times that these old commands were used to specify. It converts as close as possible to the value you specify.
This setting is volatile, so it will need to be set again each time the drive is power cycled (shutdown the PC and power it back on).

For old drives, the best way to specify a time to spindown is with `--standby` as APM does not specify exactly when a drive should spin down and it is up to the vendor to define this timer.

APM was added to the spec to allow a system to specify more detail about how it wants to use the device using a sliding scale between 1 and 255 to save more power or provide more performance. APM does not specify what the timer values should be set to and this is up to the vendor to set in their firmware. APM does specify that values 128-255 prohibit spinning down for better performance.
One important note about APM is that it is independent of the standby timer described above.
If the APM feature is enabled and the standby timer is enabled (has been set by the command above), the drive will only go into standby if:

1. the standby timer expires, or
2. the vendor specific APM algorithm indicates the standby state should be entered.

APM is also a volatile feature so power cycling the drive will bring it back to defaults again.

The `--standby` option is more or less equivalent to EPC's `--standby_z` option.
EPC has interactions with this old method of setting the timer as well as interactions with APM.
The APM interaction is basically you can only choose one: EPC or APM to use at a given time. You cannot use both EPC and APM at the same time.

EPC will honor the changes by the old `--standby` commands as temporary changes.
If you have EPC and want the change to stay across power cycles, use the `--standby_z` option as it will save all changes by default.
